src/sentry/static/sentry/app/components/events/interfaces/threads/getThreadStacktrace.ts

```diff
@@ -0,0 +1,30 @@
+import {Thread} from 'app/types/events';
+import {Event} from 'app/types';
+
+import getThreadException from './getThreadException';
+
+// TODO(ts): define raw type
+function getThreadStacktrace(thread: Thread, event: Event, raw?: any) {
+  const exc = getThreadException(thread, event);
+  if (exc) {
+    let rv = null;
+    for (const singleExc of exc.values) {
+      if (singleExc.threadId === thread.id) {
+        rv = (raw && singleExc.rawStacktrace) || singleExc.stacktrace;
+      }
+    }
+    return rv;
+  }
+
+  if (raw && thread.rawStacktrace) {
+    return thread.rawStacktrace;
+  }
+
+  if (thread.stacktrace) {
+    return thread.stacktrace;
+  }
+
+  return null;
```

Comment:
Should you return `undefined | value` you have a mix of undefined & null being used as the 'no value' type.

Reply:
I didn't want to change this code..I've just moved it, but yes you're right :)